Helper/AdapterHelper.php

```diff
@@ -298,16 +104,16 @@ public function isAllowed()
         $storeId = $this->getStoreId();
 
         return
-            $this->config->getApplicationID($storeId)
-            && $this->config->getAPIKey($storeId)
-            && $this->config->isEnabledFrontEnd($storeId)
-            && $this->config->makeSeoRequest($storeId);
+            $this->algoliaHelper->getConfigHelper()->getApplicationID($storeId)
+            && $this->algoliaHelper->getConfigHelper()->getAPIKey($storeId)
+            && $this->algoliaHelper->getConfigHelper()->isEnabledFrontEnd($storeId)
+            && $this->algoliaHelper->getConfigHelper()->makeSeoRequest($storeId);
     }
 
     /** @return bool */
     public function isSearch()
     {
-        return $this->request->getFullActionName() === 'catalogsearch_result_index';
+        return $this->filtersHelper->getRequest()->getFullActionName() === 'catalogsearch_result_index';
```

Comment:
The same as before applies here as well. `filterHelper` should be responsible only to count / prepare filters, but not provision `request`. WDYT of having `CatalogSearch helper` to provision it? Or do you have any other idea?